### PR TITLE
python3Packages.gudhi: 3.8.0 -> 3.9.0

### PR DESCRIPTION
**Pull request by yl3dy**
## Description of changes

Updated to the latest stable, changed format to the more appropriate "pyproject", replaced a patch with a simple substitution.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/gudhi/default.nix [hunk 1]
```diff
@@ -19,20 +20,18 @@
 
 buildPythonPackage rec {
   pname = "gudhi";
-  version = "3.8.0";
-  format = "setuptools";
+  version = "3.9.0";
+  format = "pyproject";
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/gudhi/default.nix:24]: afaicr we now use
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/gudhi/default.nix [hunk 2]
```diff
@@ -43,13 +42,22 @@ buildPythonPackage rec {
...
   ];
 
+  prePatch = ''
+    substituteInPlace src/python/CMakeLists.txt \
+      --replace '"''${GUDHI_PYTHON_PATH_ENV}"' ""
+  '';
+
   preBuild = ''
     cd src/python
   '';
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/gudhi/default.nix:52]: Usually we set 
```suggestion
  sourceRoot = "${src.name}/src/python";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/gudhi/default.nix (hunk 2 above)
Comment (yl3dy) [reply pkgs/development/python-modules/gudhi/default.nix:52]: The build system depends on `CMakeLists.txt` in the root of the project, so unfortunately custom `sourceRoot` breaks the build.

File: pkgs/development/python-modules/gudhi/default.nix (hunk 1 above)
Comment (yl3dy) [reply pkgs/development/python-modules/gudhi/default.nix]: Yeah, it seems so. Fixed!

File: pkgs/development/python-modules/gudhi/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/development/python-modules/gudhi/default.nix:52]: OK.

File: pkgs/development/python-modules/gudhi/default.nix [hunk 3]
```diff
@@ -43,13 +42,22 @@ buildPythonPackage rec {
     "-DPython_ADDITIONAL_VERSIONS=3"
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/gudhi/default.nix:42]: Use `lib.cmake*`

**Review by AndersonTorres — APPROVED** [review]
Besides, lgtm

File: pkgs/development/python-modules/gudhi/default.nix (hunk 3 above)
Comment (yl3dy) [reply pkgs/development/python-modules/gudhi/default.nix]: Done!

File: pkgs/development/python-modules/gudhi/default.nix [hunk 4]
```diff
@@ -19,37 +20,44 @@
...
+
   preBuild = ''
     cd src/python
   '';
 
   checkPhase = ''
+    runHook preCheck
+
     rm -r gudhi
     ${cmake}/bin/ctest --output-on-failure
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gudhi/default.nix:58]: To fix cross:

```suggestion
    ${buildPackages.cmake}/bin/ctest --output-on-failure
```
Also don't we have cmake in nativeBuildInputs already?

```suggestion
    ctest --output-on-failure
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/gudhi/default.nix]: buildPackages is the pkgs for the host the build is done on. For nativeBuildInputs that is automatically done.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
@ofborg build python3Packages.gudhi

**Pull request merged**